llvm/lib/YkIR/YkIRWriter.cpp

```diff
@@ -388,6 +389,26 @@ class YkIRWriter {
     InstIdx++;
   }
 
+  void serialiseDeoptSafepointInst(CallInst *I, ValueLoweringMap &VLMap, unsigned BBIdx,
+                         unsigned &InstIdx) {
+      serialiseOpcode(OpCodeDeoptSafepoint);
+      // stackmap ID:
+      serialiseOperand(I, VLMap, I->getOperand(0));
+
+      // num_shadow_bytes:
+      serialiseOperand(I, VLMap, I->getOperand(1));
+
+      // num_lives:
+      OutStreamer.emitInt32(I->arg_size() - 2);
```

Comment:
Would it be better to fetch the argument counting from the beginning (instead of the end)?

(I guess arg_size() doesn't return the varargs args (in this case the live vars?))

Reply:
> Would it be better to fetch the argument counting from the beginning (instead of the end)?

What do you mean? I copied this from the `serialisedCallInst` and there is a comment there about how `I->arg_size()` includes both static and varargs.

Reply:
I'm an idiot. Yes, this is correct. The number of args minus the id and shadow.